browser/src/react-redux/App/SignupPage/SignupPage.react.js

```diff
@@ -69,77 +75,71 @@ class SignupPage extends Component {
         if (err.response.data.error.errors) {
           const errors = err.response.data.error.errors;
 
-          errors.email ? this.setState({ emailError: errors.email.message }) : null
-          errors.password ? this.setState({ passwordError: errors.password.message }) : null
-
+          errors.email ? this.setState({ emailError: errors.email.message }) : null;
+          errors.password ? this.setState({ passwordError: errors.password.message }) : null;
         } else {
           this.setState({
-            generalError: err.response.data.error.message
+            generalError: err.response.data.error.message,
           });
         }
       });
-  }
+  };
   componentDidUpdate(prevProps) {
     if (this.props.user !== prevProps.user) {
-      this.props.history.push('/')
+      this.props.history.push('/');
```

Comment:
This won't trigger when we initially load the component. This would require checks in `componentDidUpdate` and `componentDidMount`. Why not just load something like `<Redirect>` from `react-router-dom` in the render if the user exists? That would cover all cases.